azure/converters/subnets.go

```diff
@@ -17,11 +17,27 @@ limitations under the License.
 package converters
 
 import (
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v4"
 	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-08-01/network"
 	"k8s.io/utils/ptr"
 	"sigs.k8s.io/cluster-api-provider-azure/azure"
 )
 
+// GetSubnetAddressesV2 returns the address prefixes contained in a subnet.
+func GetSubnetAddressesV2(subnet *armnetwork.Subnet) []string {
+	var addresses []string
+	if subnet.Properties != nil && subnet.Properties.AddressPrefix != nil {
+		addresses = []string{ptr.Deref(subnet.Properties.AddressPrefix, "")}
+	} else if subnet.Properties != nil && subnet.Properties.AddressPrefixes != nil {
+		for _, prefix := range subnet.Properties.AddressPrefixes {
+			if ptr.Deref(prefix, "") != "" {
```

Comment:
nit: I wonder if it might match the previous behavior a tad more closely if we only check that the value here isn't nil:
```suggestion
			if prefix != nil {
```
I think the only case this affects is where `prefix` is a non-nil, empty string. I'm not sure if that scenario is even possible though.